ide/languages.hcl/src/org/netbeans/modules/languages/hcl/ast/HCLResolveOperation.java

```diff
@@ -28,22 +28,30 @@ public sealed interface HCLResolveOperation extends HCLExpression {
 
 
     HCLExpression base();
+
     default List<? extends HCLExpression> elements() {
             return List.of(base());
     }
 
     public record Attribute(HCLExpression base, HCLIdentifier attr) implements HCLResolveOperation {
+
+        public Attribute {
+            if (base == null) {
+                throw new NullPointerException("base cannot be null");
+            }
+        }
+
         @Override
         public String asString() {
-            return base.asString() + "." + attr;
+            return HCLExpression.asString(base) + "." + HCLExpression.asString(attr);
```

Comment:
can use `base.asString()` since base can't be null due to constructor validation. (3x more occurrences in this class)